File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    answer = -1
    
    while 1:
        pre = arr
        arr = [n/2 if 50 <= n and n%2==0 else n*2+1 if n < 50 and n%2==1 else n for n in pre]
        answer += 1
        if pre == arr: break
        
    return answer

File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.Arrays;

class Solution {
    public int solution(int[] arr) {
        int answer = -1;
        int[] pre = new int[arr.length];
        int[] next = arr;
        
        do {
            pre = Arrays.copyOf(next, next.length);
            next = func(arr);
            answer++;    
        } while (Arrays.compare(pre, next) != 0);
            
        return answer;
    }
    
    public int[] func(int[] arr) {
        for (int i=0; i<arr.length; i++) {
            if (50 <= arr[i] && arr[i]%2==0) {
                arr[i] /= 2;
            } else if (arr[i] < 50 && arr[i]%2==1) {
                arr[i] = arr[i]*2 + 1;
            }
        }
        return arr;
    }
}